db/migrate/20210611111537_create_users.rb

```diff
@@ -0,0 +1,11 @@
+class CreateUsers < ActiveRecord::Migration[6.1]
+  def change
+    create_table :users do |t|
+      t.string :username,        default: '', null: false
+      t.string :email,           default: '', null: false
```

Comment:
Please describe why you uses empty text instead of null value?

Reply:
I've done it like this, because I've done it previously this way, but there is no such a big difference, but if we want to use raw queries it would be better to have `null` than `''`